UnityWeld/Binding/Internal/AdapterResolver.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using UnityWeld.Ioc;
+
+namespace UnityWeld.Binding.Internal
+{
+    /// <summary>
+    /// Helper class for creating adapters
+    /// </summary>
+    public static class AdapterResolver
+    {
+        private static IWeldContainerIoC container;
+
+        public static IAdapter CreateAdapter(Type adapterType)
+        {
+            if(container == null)
+            {
+                SetupWeldContainer();
+            }
+            return container.Resolve<IAdapter>(adapterType);
+        }
+
+        private static void SetupWeldContainer()
+        {
+            var containerTypes = TypeResolver.TypesWithWeldContainerAttribute.Where(t => typeof(IWeldContainerIoC).IsAssignableFrom(t));
```

Comment:
Should call `.ToList()` on this to avoid multiple enumeration. If it is a list, the later call to Linq `.Count()` can be replaced with just the `.Count` property, which is O(1) instead of O(n) complexity.